tests/upgrades/conftest.py

```diff
@@ -342,11 +352,20 @@ def pytest_collection_modifyitems(items, config):
                 depend_on_node_ids = [_get_test_node_id(f) for f in dependant_on_functions]
                 for depend_on_node_id in depend_on_node_ids:
                     if depend_on_node_id in pre_upgrade_failed_tests:
-                        log(
-                            f'Deselected (because of dependant test failed): {item.nodeid}',
-                            'INFO',
-                        )
-                        deselected_items.append(item)
-
-        config.hook.pytest_deselected(items=deselected_items)
-        items[:] = [item for item in items if item not in deselected_items]
+                        item.add_marker(
+                            pytest.mark.fail(
+                                reason=f'The dependant test {item.nodeid} failed!'))
+
+
+class DependentTestFailed(Exception):
+    """Raise when the dependent test fails"""
+    pass
+
+
+@pytest.fixture(autouse=True)
+def fail_xfail_test(request):
```

Comment:
Let's give this a more descriptive name, since `xfail` isn't actually in use here or what this fixture is doing. Consider `post_upgrade_dependant_fail`